unlock-app/src/hooks/useCheckoutCommunication.ts

```diff
@@ -0,0 +1,43 @@
+import { usePostmateParent } from './usePostmateParent'
+
+export interface UserInfo {
+  address: string
+}
+
+export interface TransactionInfo {
+  hash: string
+}
+
+export enum CheckoutEvents {
+  userInfo = 'checkout.userInfo',
+  closeModal = 'checkout.closeModal',
+  transactionInfo = 'checkout.transactionInfo',
+}
+
+// This is just a convenience hook that wraps the `emit` function
+// provided by the parent around some communication helpers. Consumers
+// are intended to check the `ready` property to know when to start
+// using the hook. If they do not, runtime errors may occur if any of
+// these functions are called before the postmate handshake completes.
+export const useCheckoutCommunication = () => {
```

Comment:
Probably Overkill but would there be a way to "cache" the calls until ready if but ready yet?

Reply:
We actually do something like that in the main window side of the current paywall: https://github.com/unlock-protocol/unlock/blob/c00cb0655289eec82a5f1f1e2a5e38f7e7a257ad/paywall/src/unlock.js/PostMessageEmitters/CheckoutIframeMessageEmitter.ts#L65

It's doable, but probably a bit more slippery when we're doing it in/with a hook

Reply:
I think it makes sense as a future enhancement

Reply:
Thanks